src/Elastic.Apm.AspNetFullFramework/ApplicationConfigurationReader.cs

```diff
@@ -33,7 +39,7 @@ private string FindAppPoolName()
 			{
 				return Environment.GetEnvironmentVariable("APP_POOL_ID");
 			}
-			catch (Exception ex)
+			catch (SecurityException ex)
```

Comment:
Why is this necessary? I think the catch block is a very generic error handler - even if the current `Environment.GetEnvironmentVariable` only throws `SecurityException` I don't see why it's bad to have a general error handler for every future case.

Reply:
Catch-all is very bad practice and generally avoided and recommended against almost ubiquitously, and for very good reason. If you are unaware of the exceptions that can be expected, then this is a bad situation to be in because you are implemented generic handling of unexpected exceptions. You are treating all possible unexpected exceptions as a security exception would be treated. This can result in silent errors, or bad object states.

Unit tests should cover all possible exceptions, so if any future code is added that could generate an exception, this should be caught in unit tests and then handled accordingly. Adding a catch-all here is potentially dangerous, bad practice, and in relation to your reply, breaks YAGNI.